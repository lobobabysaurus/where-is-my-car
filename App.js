import React from 'react'
import { Alert, View } from 'react-native'

import { getCoords, saveCoords } from './lib/coords'
import { MarkButton, MarkedMap } from './lib/components'
import style from './lib/styles'

export default class App extends React.Component {

  region = {
    latitude: 39.973960,
    latitudeDelta: 0.0111,
    longitude: -75.130620,
    longitudeDelta: 0.0111,
  }

  state = {
    allCoords: [],
    markedCoords: undefined,
  }

  componentDidMount = async () => {
    const coords = await getCoords()
    if (coords) {
      this._updateCoordinates(coords)
    }
  }

  render() {
    return (
      <View style={style.container}>
        <MarkedMap coords={this.state.markedCoords} region={this.region}/>
        <MarkButton onPress={this._trackLocation} />
      </View>
    )
  }

  _trackLocation = () => {
    navigator.geolocation.getCurrentPosition(
      async (position) => {
        const coords = await saveCoords(position.coords)
        this._updateCoordinates(coords)
      },
      (err) => Alert.alert(`Error getting location: ${err}`)
    )
  }

  _updateCoordinates = (coords) => {
    this.setState({
      allCoords: coords,
      markedCoords: coords[coords.length-1]
    })
  }
}
