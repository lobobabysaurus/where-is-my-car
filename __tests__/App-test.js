import React from 'react'
import renderer from 'react-test-renderer'

import App from '../App'

jest.mock('react-native-maps', () => {
  const React = require.requireActual('react')
  const MapView = require.requireActual('react-native-maps')

  const MockCallout = (props) => (
    React.createElement('Callout', props, props.children)
  )

  const MockMarker = (props) => (
    React.createElement('Marker', props, props.children)
  )

  const MockMapView = (props) => (
    React.createElement('MapView', props, props.children)
  )

  MockCallout.propTypes = MapView.Callout.propTypes
  MockMarker.propTypes = MapView.Marker.propTypes
  MockMapView.propTypes = MapView.propTypes
  MockMapView.Marker = MockMarker
  MockMapView.Callout = MockCallout
  return MockMapView
})


describe('App', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<App />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
