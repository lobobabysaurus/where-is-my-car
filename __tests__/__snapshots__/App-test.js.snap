// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<View>
  <MapView
    initialRegion={
      Object {
        "latitude": 39.97396,
        "latitudeDelta": 0.0111,
        "longitude": -75.13062,
        "longitudeDelta": 0.0111,
      }
    }
    style={
      Object {
        "height": "90%",
        "width": "100%",
      }
    }
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "purple",
        "height": "10%",
        "justifyContent": "space-evenly",
        "width": "100%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 36,
        }
      }
    >
      Mark
    </Text>
  </View>
</View>
`;
